org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/nexus/NexusScanFileManager.java

```diff
@@ -234,23 +234,20 @@ protected SolsticeScanMonitor createSolsticeScanMonitor(ScanModel model) {
 		return solsticeScanMonitor;
 	}
 
+	@SuppressWarnings("squid:S1452")
 	protected Map<ScanRole, Collection<INexusDevice<?>>> extractNexusDevices(ScanModel model) throws ScanningException {
 
 		Map<ScanRole, Collection<INexusDevice<?>>> nexusDevices = new EnumMap<>(ScanRole.class);
 		nexusDevices.put(ScanRole.DETECTOR,  getNexusDevices(model.getDetectors()));
 		nexusDevices.put(ScanRole.SCANNABLE, getNexusDevices(model.getScannables()));
-
-		if (model.getMonitors()!=null) {
-			Collection<IScannable<?>> perPoint = model.getMonitors().stream().filter(scannable -> scannable.getMonitorRole()==MonitorRole.PER_POINT).collect(Collectors.toList());
-			Collection<IScannable<?>> perScan  = model.getMonitors().stream().filter(scannable -> scannable.getMonitorRole()==MonitorRole.PER_SCAN).collect(Collectors.toList());
-			nexusDevices.put(ScanRole.MONITOR_PER_POINT,   getNexusDevices(perPoint));
-			nexusDevices.put(ScanRole.MONITOR_PER_SCAN,  getNexusDevices(perScan));
-		}
+		nexusDevices.put(ScanRole.MONITOR_PER_POINT, getNexusDevices(model.getMonitorsPerPoint()));
+		nexusDevices.put(ScanRole.MONITOR_PER_SCAN, getNexusDevices(model.getMonitorsPerScan()));
 		nexusDevices.put(ScanRole.NONE, Collections.emptyList());
 
 		return nexusDevices;
 	}
 
+	@SuppressWarnings({"squid:S1452", "squid:S3776"})
 	protected Map<ScanRole, List<NexusObjectProvider<?>>> extractNexusProviders() throws ScanningException {
 		Map<ScanRole, List<NexusObjectProvider<?>>> nexusObjectProviders = new EnumMap<>(ScanRole.class);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "nexusObjectProviders" which hides the field declared at line 93. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)